From 39079bbc941969cfa2fc643f93d322bc7e156530 Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Sun, 6 Jun 2021 04:31:54 +0530
Subject: [PATCH 060/212] hack! arm64: dts: xiaomi-beryllium-common: Add nodes
 for q6voice apr service

Original author: Joel Selvaraj <jo@jsfamily.in>  # removed from author
to prevent checkpatch complaining

Signed-off-by: David Heidelberg <david@ixit.cz>
---
 .../qcom/sdm845-xiaomi-beryllium-common.dtsi  | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
index 796ab30d1b85..f8b3e30b904c 100644
--- a/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm845-xiaomi-beryllium-common.dtsi
@@ -8,6 +8,9 @@
 #include <dt-bindings/regulator/qcom,rpmh-regulator.h>
 #include <dt-bindings/sound/qcom,q6afe.h>
 #include <dt-bindings/sound/qcom,q6asm.h>
+#include <dt-bindings/sound/qcom,q6afe.h>
+#include <dt-bindings/sound/qcom,q6asm.h>
+#include <dt-bindings/sound/qcom,q6voice.h>
 #include "sdm845.dtsi"
 #include "sdm845-wcd9340.dtsi"
 #include "pm8998.dtsi"
@@ -520,6 +523,13 @@ cpu {
 		};
 	};
 
+	dai-link-voicemmode1 {
+		link-name = "VoiceMMode1";
+		cpu {
+			sound-dai = <&q6voicedai VOICEMMODE1>;
+		};
+	};
+
 	tas2559-dai-link {
 		link-name = "Primary Spkr Playback";
 
@@ -719,3 +729,15 @@ &wifi {
 	qcom,snoc-host-cap-skip-quirk;
 };
 
+/* qdsp6 hacks */
+&q6cvp {
+	status = "okay";
+};
+
+&q6cvs {
+	status = "okay";
+};
+
+&q6mvm {
+	status = "okay";
+};
-- 
2.55.0

